src/components/pdf-upload/PdfUpload.tsx

```diff
@@ -137,11 +148,28 @@ export const PdfUpload = ({
         </div>
       )}
 
+      {/* Filename Input */}
+      <div className="space-y-3">
+        <label
+          htmlFor="pdf-filename"
+          className="text-ob-base-300 text-sm font-medium mb-2 block"
+        >
+          Filename
+        </label>
+        <Input
+          id="pdf-filename"
+          placeholder="Enter a filename for this PDF..."
+          value={filename}
+          onValueChange={(value) => setFilename(value)}
+          disabled={loading || !selectedFile}
+        />
+      </div>
+
       {/* Description Input */}
-      <div className="space-y-2">
+      <div className="space-y-3">
         <label
           htmlFor="pdf-description"
-          className="text-ob-base-300 text-sm font-medium"
+          className="text-ob-base-300 text-sm font-medium mb-2 block"
         >
           Description (optional)
```

Comment:
i actually don't like this field at all. no need to remove now, but it's so unstructured and seems unlikely to be useful. i feel like the AI should generate whatever descriptions are generated